### PR TITLE
move trigger tab to react

### PR DESCRIPTION
**Pull request by sunilyadav840**
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/855?feature.someFeatureFlagYouMightNeed=true)

Migration of trigger Tab into react and remove knockout TriggerTab.ts and TriggerTab.html file

https://user-images.githubusercontent.com/79906609/120454077-2f973b00-c3b1-11eb-84b8-8afd633733ea.mp4

https://user-images.githubusercontent.com/79906609/121141252-66110200-c858-11eb-991b-dd193d3d1ef5.mp4

### CODE REVIEW

File: src/Explorer/Tabs/TriggerTabContent.tsx [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+    const { triggerId, triggerBody, triggerOperation, triggerType } = this.state;
+    return updateTrigger(this.props.collection.databaseId, this.props.collection.id(), {
+      id: triggerId,
+      body: triggerBody,
+      triggerOperation: triggerOperation as SqlTriggerResource["triggerOperation"],
+      triggerType: triggerType as SqlTriggerResource["triggerType"],
+    })
+      .then(
+        (createdResource) => {
```
Comment (southpolesteve) [inline src/Explorer/Tabs/TriggerTabContent.tsx:151]: Switch this to use async/await

**Comment by southpolesteve** [comment]
Bug: After saving a Tigger, the ID should no longer be editable

**Comment by southpolesteve** [comment]
Bug: Viewing an existing trigger, the ID should not be editable

File: src/Explorer/Tabs/TriggerTabContent.tsx (hunk 1 above)
Comment (sunilyadav840) [reply src/Explorer/Tabs/TriggerTabContent.tsx]: Updated code

File: src/Explorer/Tree/Trigger.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export default class Trigger {
     });
   }
 
-  public static create(source: ViewModels.Collection, event: MouseEvent) {
+  public static create(container: Explorer, source: ViewModels.Collection, event: MouseEvent) {
```
[the remark this answers is not in the document]
Comment (sunilyadav840) [reply src/Explorer/Tree/Trigger.ts]: Removed.

**Comment by sunilyadav840** [comment]
@southpolesteve 

> Bug: After saving a Tigger, the ID should no longer be editable

Fixed the bug. Added video link in description for reference.

**Comment by sunilyadav840** [comment]
> Bug: Viewing an existing trigger, the ID should not be editable

Fixed the bug. Added video link in description for reference

**Comment by victor-meng** [comment]
The browser tab starts to flicker after opening the trigger tab.

https://user-images.githubusercontent.com/56978073/121273793-e7c75680-c87d-11eb-8a5a-e2fcd1ccecd0.mp4

File: src/Explorer/Tabs/TriggerTabContent.tsx [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+            label="Trigger Id"
+            id="entityTimeId"
+            autoFocus
+            required
+            type="text"
+            pattern="[^/?#\\]*[^/?# \\]"
+            placeholder="Enter the new trigger id"
+            size={40}
+            value={triggerId}
+            disabled={!isIdEditable}
```
Comment (victor-meng) [inline src/Explorer/Tabs/TriggerTabContent.tsx:310]: Use `readOnly` instead of `disabled`.

Comment (sunilyadav840) [reply src/Explorer/Tabs/TriggerTabContent.tsx]: Changes updated

**Comment by sunilyadav840** [comment]
> The browser tab starts to flicker after opening the trigger tab. Check out 0:32 and 0:38 in the video below.
> 
>  vimeng-sql-test.-.Microsoft.Azure.-.Google.Chrome.2021-06-08.17-19-53.mp4 
> After this icon shows up in the editor, the tab becomes very laggy. If I leave the tab open for some time then the browser tab would eventually crash.
> image

Thank you @victor-meng  for mentioning issue.
It is fixed and code pushed. 
Uploaded video for reference.

https://user-images.githubusercontent.com/79906609/121860089-22b20a00-cd16-11eb-8b3d-d9e95fa79e6d.mp4

**Pull request merged**